tools/src/publish-packages/tasks/commentOnIssuesTask.ts

```diff
@@ -0,0 +1,199 @@
+ */
+export const commentOnIssuesTask = new Task<TaskArgs>(
+  {
+    name: 'commentOnIssuesTask',
+    dependsOn: [selectPackagesToPublish],
+    backupable: true,
+  },
+  async (parcels: Parcel[], options: CommandOptions) => {
+    logger.info('\n🐙 Commenting on issues closed by published changes');
+
+    const payload = await generatePayloadForCommentatorAsync(parcels, options.tag);
+
+    if (!payload.length) {
+      logger.log('There are no closed issues to comment on\n');
+      return;
+    }
+    if (options.dry) {
+      logger.debug('Skipping due to --dry flag');
+      logManualFallback(payload);
+      return;
+    }
+    if (!process.env.GITHUB_TOKEN) {
+      logger.error(
+        'Environment variable `%s` must be set to dispatch a commentator workflow',
+        chalk.magenta('GITHUB_TOKEN')
+      );
+      logManualFallback(payload);
+      return;
+    }
+
+    const currentBranchName = await Git.getCurrentBranchNameAsync();
+
+    // Sometimes we publish from different branches (especially for testing) where comments are not advisable.
+    if (currentBranchName !== 'master') {
+      logger.warn('This feature is disabled on branches other than master');
+      logManualFallback(payload);
+      return;
+    }
+
+    // Dispatch commentator workflow on GitHub Actions with stringified and escaped payload.
+    await dispatchWorkflowEventAsync('commentator.yml', currentBranchName, {
+      payload: JSON.stringify(payload).replace(/("|`)/g, '\\$1'),
+    });
+    logger.success(
+      'Successfully dispatched commentator action for the following issues: %s',
+      linksToClosedIssues(payload.map(({ issue }) => issue))
+    );
+  }
+);
+
+/**
+ * Generates payload for `expotools commentator` command.
+ */
+async function generatePayloadForCommentatorAsync(
+  parcels: Parcel[],
+  tag: string
+): Promise<CommentatorPayload> {
+  // An object whose key is the issue number and value is an array of rows to put in the comment's body.
+  const commentRows: Record<number, CommentRowObject[]> = {};
+
+  // An object whose key is the pull request number and value is an array of issues it closes.
+  const closedIssuesRegistry: Record<number, number[]> = {};
+
+  for (const { pkg, state } of parcels) {
+    const versionChanges = state.changelogChanges?.versions[UNPUBLISHED_VERSION_NAME];
+
+    if (!versionChanges) {
+      continue;
+    }
+    const allEntries = ([] as ChangelogEntry[]).concat(...Object.values(versionChanges));
+    const allPullRequests = new Set(
+      ([] as number[]).concat(...allEntries.map((entry) => entry.pullRequests ?? []))
+    );
+
+    // Visit all pull requests mentioned in the changelog.
+    for (const pullRequest of allPullRequests) {
+      // Look for closed issues just once per pull request to reduce number of GitHub API calls.
+      if (!closedIssuesRegistry[pullRequest]) {
+        closedIssuesRegistry[pullRequest] = await getClosedIssuesAsync(pullRequest);
+      }
```

Comment:
Good idea 👍 We should definitely do this for all requests to GitHub API calls, so let me add this in a separate PR. Right now I implemented that only for this one request because this one needs to be performed most often — especially when we publish all packages where each has a reference to the same pull request (like dropping support of iOS version).